workflow-service/src/test/java/com/redhat/parodos/project/controller/ProjectControllerTest.java

```diff
@@ -0,0 +1,146 @@
+		ProjectResponseDTO project2DTO = getSampleProject("project2");
+		Mockito.when(projectService.getProjects()).thenReturn(List.of(project1DTO, project2DTO));
+
+		// When
+		mockMvc.perform(this.getRequestWithValidCredentials("/api/v1/projects/"))
+				.andExpect(MockMvcResultMatchers.status().isOk())
+				.andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].id", Matchers.is(project1DTO.getId())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[0].name", Matchers.is(project1DTO.getName())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].id", Matchers.is(project2DTO.getId())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$[1].name", Matchers.is(project2DTO.getName())));
+
+		// Then
+		Mockito.verify(projectService, Mockito.times(1)).getProjects();
+	}
+
+	@Test
+	public void testGetProjectsWithInvalidCredentials() throws Exception {
+		// When
+		mockMvc.perform(this.getRequestWithInValidCredentials("/api/v1/projects/"))
+				.andExpect(MockMvcResultMatchers.status().isUnauthorized());
+		// Then
+		Mockito.verify(projectService, Mockito.times(0)).getProjects();
+	}
+
+	@Test
+	public void testGetProjectbyIdWithValidID() throws Exception {
+		ProjectResponseDTO project1DTO = getSampleProject("project1");
+		Mockito.when(projectService.getProjectById(UUID.fromString(project1DTO.getId()))).thenReturn(project1DTO);
+
+		// When
+		mockMvc.perform(this.getRequestWithValidCredentials(String.format("/api/v1/projects/%s", project1DTO.getId())))
+				.andExpect(MockMvcResultMatchers.status().isOk())
+				.andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+				.andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(project1DTO.getId())))
+				.andExpect(MockMvcResultMatchers.jsonPath("$.name", Matchers.is(project1DTO.getName())));
+
+		// Then
+		Mockito.verify(projectService, Mockito.times(1)).getProjectById(Mockito.any());
+	}
+
+	@Test
+	public void testGetProjectbyIdWithInValidID() throws Exception {
+		ProjectResponseDTO project1DTO = getSampleProject("project1");
+		Mockito.when(projectService.getProjectById(Mockito.any())).thenReturn(null);
+
+		// When
+		mockMvc.perform(this.getRequestWithValidCredentials(String.format("/api/v1/projects/%s", UUID.randomUUID())))
+				.andExpect(MockMvcResultMatchers.status().isNotFound());
+		// Then
+		Mockito.verify(projectService, Mockito.times(1)).getProjectById(Mockito.any());
+	}
+
+	@Test
+	public void testGetProjectbyIdWithValidIDWithoutRightCredentials() throws Exception {
+		// When
+		mockMvc.perform(this.getRequestWithInValidCredentials(String.format("/api/v1/projects/%s", UUID.randomUUID())))
+				.andExpect(MockMvcResultMatchers.status().isUnauthorized());
+		// Then
+		Mockito.verify(projectService, Mockito.times(0)).getProjectById(Mockito.any());
+
+	}
+
+	ProjectResponseDTO getSampleProject(String name) {
```

Comment:
I don't like this method name. At the first glance, it seems that invoke some API to retrieve the info, so I felt the need to check the implementation. Do you think It's possible to improve it?
For example, `createProjectResponseDTO` or `createTestSampleProject`?